### PR TITLE
#158581899 Alter all rooms query to to filter by location

### PR DESCRIPTION
**Pull request by dkam26**
#### What does this PR do?

   When the location parameter is provided, the query should  only return rooms in that location:
    else it should return all rooms

#### What are the relevant pivotal tracker stories?

    #158581899

#### Screenshots 
   
     When a location parameter is provided.

screen shot 2018-07-03 at 12 09 16 pm

  When no location is provided.

screen shot 2018-07-03 at 1 24 33 pm

### CODE REVIEW

File: api/room/models.py [hunk 1]
```diff
@@ -14,8 +15,10 @@ class Room(Base, Utility):
     capacity = Column(Integer, nullable=False)
     image_url = Column(String)
     calendar_id = Column(String)
+    location = Column(String, ForeignKey('locations.name'))
     floor_id = Column(Integer, ForeignKey('floors.id'))
     floor = relationship('Floor')
+    locations = relationship('Location')
```
Comment (PatrickCmd) [inline api/room/models.py]: change this field to singular **location** instead of **locations**

**Comment by dkam26** [comment]
@proxiex ,all changes were made.

File: api/room/schema.py [hunk 2]
```diff
@@ -45,6 +46,7 @@ class Arguments:
         capacity = graphene.Int()
         image_url = graphene.String()
         calendar_id = graphene.String()
+        location = graphene.String()
```
Comment (shemogumbe) [inline api/room/schema.py]: Any reason why the location field was added, thought this was to be found by following back the foreign key relationship?

File: api/room/models.py [hunk 3]
```diff
@@ -18,7 +18,7 @@ class Room(Base, Utility):
     location = Column(String, ForeignKey('locations.name'))
     floor_id = Column(Integer, ForeignKey('floors.id'))
     floor = relationship('Floor')
-    locations = relationship('Location')
+    location_name = relationship('Location')
```
Comment (shemogumbe) [inline api/room/models.py]: Any reason for specifically calling this 'location_name' and not location to reference the table Location?

**Pull request closed without merging**